src/pages/[teamName]/[projectName]/[testId].tsx

```diff
@@ -207,13 +208,28 @@ const TestPage = withError<GET_SERVER_SIDE_PROPS_ERROR, ITestProps>(
         {
           colorMode: useColorMode().colorMode,
           logs: versionTriage(JSON.parse(log)),
-          priority: {},
+          index: useState(1),
         },
         (p) => ({
           ...p,
           failures: p.logs.commands.filter((a) => a.success === false),
         }),
-        ({ colorMode, logs, failures }) => (
+        (p) => ({
+          ...p,
+          restFailures: p.failures.filter(
+            (b) => b.exchange[0].meta.apiType === "rest"
+          ),
+          graphqlFailures: p.failures.filter(
+            (b) => b.exchange[0].meta.apiType === "graphql"
+          ),
+        }),
+        ({
+          colorMode,
+          logs,
+          restFailures,
+          graphqlFailures,
+          index: [index, setIndex],
```

Comment:
Currently, this `setIndex` is destructured but not used. If you want the changes in `index` to be reflected from `SegmentedControl` to `[testId].tsx`, you have to pass `setIndex` as an argument to `SegmentedControl` and then invoke that in addition to the local `setIndex` (I've added a comment to this review saying where this would need to happen).

Also, I'm not sure why two React hooks are needed for the index - I see that you have one in both `switch.tsx` and `[testId].tsx`. If they serve two distinct functions that's fine, but if not I'd recommend trying to consolidate them.